src/common/rtc.cc

```diff
@@ -28,6 +32,43 @@
 namespace mxnet {
 namespace rtc {
 
+namespace {
+
+  const char* gpu_arch_prefix = "--gpu-architecture=compute_";
+  const std::locale locale;
+
+  bool FindGpuArchOption(std::string option) {
+    auto lower = [&](char c) { return std::tolower(c, locale); };
+    std::transform(option.begin(), option.end(), option.begin(), lower);
+    return option.find(gpu_arch_prefix) != std::string::npos;
+  }
+
+  void SetGpuArch(std::vector<const char *>* c_options, std::string* arch_string) {
+    assert(c_options != nullptr);
+    assert(arch_string != nullptr);
+
+    // Check options for explicit device architecture information.
+    for (const auto& option : *c_options) {
+      if (FindGpuArchOption(option)) {
+        return;
+      }
+    }
+
+    // If not present, use the first gpu device present for architecture information.
+    int num_devices;
```

Comment:
It's goog practice to use {} initializer or = 0, and not leave uninitialized variables, even though in this case looks harmless.